packages/gatsby/src/schema/schema.js

```diff
@@ -175,7 +175,7 @@ const addThirdPartySchemas = ({
     const rootTypeName = fields[Object.keys(fields)[0]].type.name
     const types = schema.getTypeMap()
     Object.keys(types).forEach(typeName => {
-      if (typeName.startsWith(rootTypeName)) {
+      if (!typeName === rootTypeName) {
```

Comment:
Why change this? Third party schemas have all their types prefixed with the name of the type of the only rootquery field. We want to grab those, but not others like the builtin scalars etc.

Reply:
Good point about built-in scalars.

Third party schemas aren't prefixed, the plugin prefixes them. So we shouldn't bind plugin implementation to the core schema merging.